config/batch_inference.yaml

```diff
@@ -0,0 +1,13 @@
+ft_model_id: sayakpaul/gemma-2b-sft-qlora-no-robots
+ft_model_config_path: config/sample_config.yaml
+load_in_8bit: true
+load_in_4bit: false
+test_ds_id: sayakpaul/no_robots_only_coding
+push_lm_responses_to_hf_hub: true
+test_ds_split: test_sft
+batch_infer_data_preprocess_bs: 16
+inference_bs: 4
+repeat: 4
+lm_response_ds_id: chansung/lm_response_test
+lm_response_ds_split: batch_infer
+lm-response-append: true
```

Comment:
_(Doesn't have to be implemented)_

We should try to group together similar arguments under a broad category. Example:

```yaml
dataset_mixer:
  HuggingFaceH4/deita-10k-v0-sft: 1.0
dataset_splits:
- train_sft
- test_sft
preprocessing_num_workers: 12

# SFT trainer config
bf16: true
dataset_kwargs:
  add_special_tokens: false  # We already wrap <bos> and <eos> in the chat template
  append_concat_token: false # No need to add <eos> across samples
```

Reply:
sounds good. 

but do you know to map hierarchical structure of YAML to argparser?
any suggestions would be very helpful.

Reply:
Oh that needs to some further working actually. If you want to dig deeper here's a good reference:

* https://github.com/huggingface/alignment-handbook/blob/main/src/alignment/configs.py

This is what gets triggered here: 
https://github.com/huggingface/alignment-handbook/blob/8497caeaf1e4cbd9b0650eb730729e608d1d2d73/scripts/run_sft.py#L50

Reply:
Damm.. that looks like big amount of work. 

For now, going to suppress the parent info(group category). For instance, if there is something like below, I will simply ignore the `parent` but keep others. Hence better readable structure but no impact on the current argparse implementation. How does it sound ?

```
parent:
  child1: ...
  child2:...
```

Reply:
I think that will be more complicated instead. I am okay with the current simple structure.